### PR TITLE
ipcfg:Version the binary structure

### PR DESCRIPTION
**Pull request by davids5**
## Summary

 This adds a version for the binary file structure. 

## Impact

Should be none as this is not used yet.

## Testing

inspection.

### CODE REVIEW

File: fsutils/ipcfg/ipcfg.h [hunk 1]
```diff
@@ -49,7 +50,8 @@
 
 struct ipcfg_header_s
 {
-  uint8_t next;         /* Offset to the next IP configuration record */
+  uint8_t     next;     /* Offset to the next IP configuration record */
+  uint8_t     version;  /* For binary compatibility */
   sa_family_t type;     /* Either AF_INET or AF_INET6 */
 };
```
Comment (patacongo) [inline fsutils/ipcfg/ipcfg.h:56]: This is really insufficient.  If the version number changes, then the size will also change.  This won't work in most cases.

Comment (patacongo) [reply fsutils/ipcfg/ipcfg.h:56]: Nevermind.  I see that you ignore the data if the version number is bad.  That is okay if you don't try to read it.

**Review by patacongo — APPROVED** [review]
LGTM (ignore my other comments)

File: fsutils/ipcfg/ipcfg.h (hunk 1 above)
Comment (davids5) [reply fsutils/ipcfg/ipcfg.h:56]: Yes. The ides is that returning to factory default are better than using corrupted data.

**Pull request merged**